sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -228,4 +234,126 @@ public HttpPipeline getHttpPipeline() {
             });
     }
 
+
+    //==================================================================================================================================================
+    // Models APIs
+    //==================================================================================================================================================
+
+    /**
+     * Creates one or many models.
+     * @param models The list of models to create. Each string corresponds to exactly one model.
+     * @return A {@link PagedFlux} of created models.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ModelData> createModels(List<String> models) {
+        return new PagedFlux<>(
+            () -> withContext(context -> createModelsSinglePageAsync(models, context)),
+            nextLink -> withContext(context -> Mono.empty()));
+    }
+
+    PagedFlux<ModelData> createModels(List<String> models, Context context){
+        return new PagedFlux<>(
+            () -> createModelsSinglePageAsync(models, context),
+            nextLink -> Mono.empty());
+    }
+
+    Mono<PagedResponse<ModelData>> createModelsSinglePageAsync(List<String> models, Context context)
+    {
+        List<Object> modelsPayload = new ArrayList<>();
+        for (String model: models) {
+            try {
+                modelsPayload.add(mapper.readValue(model, Object.class));
+            }
+            catch (JsonProcessingException e) {
+                logger.error("Could not parse the model payload [%s]: %s", model, e);
+                return Mono.error(e);
+            }
+        }
+
+        return protocolLayer.getDigitalTwinModels().addWithResponseAsync(modelsPayload, context)
+            .map(
+                listResponse -> new PagedResponseBase<>(
+                    listResponse.getRequest(),
+                    listResponse.getStatusCode(),
+                    listResponse.getHeaders(),
+                    listResponse.getValue(),
+                    null,
+                    ((ResponseBase)listResponse).getDeserializedHeaders()));
+    }
+
+    /**
+     * Gets a model, including the model metadata and the model definition.
+     * @param modelId The Id of the model.
+     * @return The application/json model
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ModelData> getModel(String modelId) {
+        return withContext(context -> getModel(modelId, context));
+    }
+
+    Mono<ModelData> getModel(String modelId, Context context){
+        return protocolLayer.getDigitalTwinModels().getByIdWithResponseAsync(modelId, includeModelDefinition, context)
+            .flatMap(modelDataResponse -> Mono.just(mapper.convertValue(modelDataResponse.getValue(), ModelData.class)));
```

Comment:
The PL gives us `Mono<Response<ModelData>>, so this should be sufficient:
```java
return protocolLayer.getDigitalTwinModels().getByIdWithResponseAsync(modelId, includeModelDefinition, context)
            .flatMap(modelDataResponse -> Mono.just(modelDataResponse.getValue()));
```

Reply:
You can actually have the API in line 290 updated to call `Mono<Response<ModelData>> getModelWithResponse(String modelId, Context context)`, and get rid of 294-297 entirely

Reply:
```java
@ServiceMethod(returns = ReturnType.SINGLE)
    public Mono<ModelData> getModel(String modelId) {
        return withContext(context -> getModelWithResponse(modelId, context))
            .flatMap(response -> Mono.just(response.getValue()));
    }
```